src/transformers/models/van/modeling_van.py

```diff
@@ -0,0 +1,588 @@
+        self.drop2 = nn.Dropout(dropout_rate)
+
+
+class VanLargeKernelAttention(nn.Sequential):
+    """
+    Basic Large Kernel Attention (LKA).
+    """
+
+    def __init__(self, hidden_size: int):
+        super().__init__()
+        self.depth_wise = nn.Conv2d(hidden_size, hidden_size, kernel_size=5, padding=2, groups=hidden_size)
+        self.depth_wise_dilated = nn.Conv2d(
+            hidden_size, hidden_size, kernel_size=7, dilation=3, padding=9, groups=hidden_size
+        )
+        self.point_wise = nn.Conv2d(hidden_size, hidden_size, kernel_size=1)
+
+
+class VanLargeKernelAttentionLayer(nn.Module):
+    """
+    Computes attention using Large Kernel Attention (LKA) and attends the input.
+    """
+
+    def __init__(self, hidden_size: int):
+        super().__init__()
+        self.attention = VanLargeKernelAttention(hidden_size)
+
+    def forward(self, hidden_state):
+        attention = self.attention(hidden_state)
+        attended = hidden_state * attention
+        return attended
+
+
+class VanSpatialAttentionLayer(nn.Module):
+    """
+    Van spatial attention layer composed by projection (via conv) -> act -> Large Kernel Attention (LKA) attention ->
+    projection (via conv) + residual connetion.
+    """
+
+    def __init__(
+        self,
+        hidden_size: int,
+        hidden_act: str = "gelu",
+    ):
+        super().__init__()
+        self.pre_projection = nn.Sequential(
+            OrderedDict(
+                [
+                    ("conv", nn.Conv2d(hidden_size, hidden_size, kernel_size=1)),
+                    ("act", ACT2FN[hidden_act]),
+                ]
+            )
+        )
+        self.attention_layer = VanLargeKernelAttentionLayer(hidden_size)
+        self.post_projection = nn.Conv2d(hidden_size, hidden_size, kernel_size=1)
+
+    def forward(self, hidden_state):
+        residual = hidden_state
+        hidden_state = self.pre_projection(hidden_state)
+        hidden_state = self.attention_layer(hidden_state)
+        hidden_state = self.post_projection(hidden_state)
+        hidden_state = hidden_state + residual
+        return hidden_state
+
+
+class VanLayerScaling(nn.Module):
+    """
+    Scales the inputs by a learnable parameter initialized by `initial_value`.
+    """
+
+    def __init__(self, hidden_size: int, initial_value: float = 1e-2):
+        super().__init__()
+        self.weight = nn.Parameter(initial_value * torch.ones((hidden_size)), requires_grad=True)
+
+    def forward(self, hidden_state):
+        # unsqueezing for broadcasting
+        hidden_state = self.weight.unsqueeze(-1).unsqueeze(-1) * hidden_state
+        return hidden_state
+
+
+class VanLayer(nn.Module):
+    """
+    Van layer.
+    """
+
+    def __init__(
+        self,
+        hidden_size: int,
+        mlp_expansion: int = 4,
+        dropout_rate: float = 0.0,
+        drop_path_rate: float = 0.5,
+        hidden_act: str = "gelu",
+        layer_scale_init_value: float = 1e-2,
```

Comment:
At this stage, let's just send the config?

Reply:
Yes and no. It seems more readable that way (IMHO). Let me know what do you think, I don't know if we have a convention/"rule" about how deep we should use the config

Reply:
But you need to know the index right? Like `config.hidden_sizes[index]` to know the hidden size for this particular layer

Reply:
So it's fine imo, we did the same for SegFormer/PoolFormer

Reply:
having the index is very ugly and not informative in my opinion, passed the config and the parameters